opengrok-indexer/src/main/java/org/opengrok/indexer/web/messages/MessagesUtils.java

```diff
@@ -230,4 +231,32 @@ private static String messagesToJson(Group group, String... additionalTags) {
     public static String messagesToJson(Group group) {
         return messagesToJson(group, new String[0]);
     }
+
+    /**
+     * @return name of highest cssClass of messages present in the system or null.
+     */
+    static String getHighestCssClassLevel(Set<MessagesContainer.AcceptedMessage> messages) {
+        return messages.
+                stream().
+                map(MessagesContainer.AcceptedMessage::getCssClass).
+                map(Message.CssClassType::StringToCssClassType).
+                max(Message.CssClassType::compare).
+                map(Message.CssClassType::toString).
+                orElse(null);
+    }
+
+    /**
+     * @param tags message tags
+     * @return name of highest cssClass of messages present in the system or null.
+     */
+    public static String getCssClass(String... tags) {
+        Set<MessagesContainer.AcceptedMessage> messages = new TreeSet<>();
+        RuntimeEnvironment env = RuntimeEnvironment.getInstance();
+
+        for (String tag : tags) {
+            messages.addAll(env.getMessages(tag));
```

Comment:
sounds like a job for flatMap

Reply:
okay, I converted the code for the sake of learning stream API :-)